src/overlays/ovl_i4/fox_sz.c

```diff
@@ -120,132 +129,142 @@ void SectorZ_Missile_Update(Actor* this) {
             gPlayer[0].state_1C8 = PLAYERSTATE_1C8_LEVEL_COMPLETE;
             gPlayer[0].csState = 0;
-            gActors->state = -31072;
+            gActors[0].state = -31072;
             return;
         }
     }
 
-    if (((fabsf(this->fwork[6] - this->obj.pos.z) < 2000.0f) &&
+    // Level complete trigger check
+    if (((fabsf(this->fwork[MISSILE_TARGET_Z] - this->obj.pos.z) < 2000.0f) &&
          (((gPlayer[0].cam.eye.z < 0.0f) || (D_edisplay_801615D0.y < 0.0f)) ||
           (gPlayer[0].state_1C8 == PLAYERSTATE_1C8_GFOX_REPAIR))) &&
         (((gPlayer[0].state_1C8 == PLAYERSTATE_1C8_ACTIVE) || (gPlayer[0].state_1C8 == PLAYERSTATE_1C8_GFOX_REPAIR)) ||
          (gPlayer[0].state_1C8 == PLAYERSTATE_1C8_U_TURN))) {
         gPlayer[0].state_1C8 = PLAYERSTATE_1C8_LEVEL_COMPLETE;
         gPlayer[0].csState = 100;
-        gActors->state = -31072;
+        gActors[0].state = -31072;
     }
+    // clang-format on
 }
 
-void SectorZ_80199FCC(Actor* actor, s32 arg1) {
-    Actor_Initialize(actor);
-    actor->obj.status = OBJ_INIT;
-    actor->obj.id = OBJ_ACTOR_ALLRANGE;
-    actor->aiType = AI360_MISSILE;
+void SectorZ_SpawnMissile(ActorAllRange* this, s32 missileWaveIdx) {
+    Actor_Initialize(this);
+    this->obj.status = OBJ_INIT;
+    this->obj.id = OBJ_ACTOR_ALLRANGE;
+    this->aiType = AI360_MISSILE;
 
-    actor->obj.pos.x = D_i4_8019F4C0[arg1].x;
-    actor->obj.pos.y = D_i4_8019F4C0[arg1].y;
-    actor->obj.pos.z = D_i4_8019F4C0[arg1].z;
+    this->obj.pos.x = sMissileWaveInitPos[missileWaveIdx].x;
+    this->obj.pos.y = sMissileWaveInitPos[missileWaveIdx].y;
+    this->obj.pos.z = sMissileWaveInitPos[missileWaveIdx].z;
 
-    actor->state = 5;
-    actor->rot_0F4.y = 180.0f;
+    this->state = 5;
+    this->rot_0F4.y = 180.0f;
 
-    Object_SetInfo(&actor->info, actor->obj.id);
+    Object_SetInfo(&this->info, this->obj.id);
 
-    actor->health = 250;
-    actor->info.drawType = 2;
-    actor->info.hitbox = SEGMENTED_TO_VIRTUAL(D_SZ_6009230);
-    actor->fwork[1] = 25.0f;
-    actor->fwork[29] = 2.0f;
+    this->health = 250;
+    this->info.drawType = 2;
+    this->info.hitbox = SEGMENTED_TO_VIRTUAL(aSZMissileHitbox);
+    this->fwork[1] = 25.0f;
+    this->fwork[29] = 2.0f;
 
-    AUDIO_PLAY_SFX(NA_SE_EN_PUNCH_ENGINE, actor->sfxSource, 4);
+    AUDIO_PLAY_SFX(NA_SE_EN_PUNCH_ENGINE, this->sfxSource, 4);
 }
 
-void SectorZ_8019A0F8(Actor* actor, s32 arg1) {
-    Actor_Initialize(actor);
-    actor->obj.status = OBJ_INIT;
-    actor->obj.id = OBJ_ACTOR_ALLRANGE;
-    actor->aiType = arg1 + AI360_10 + 3;
+// Enemies that follow the missile
+void SectorZ_MissileEscort(ActorAllRange* this, s32 enemyIndex) {
+    Actor_Initialize(this);
+    this->obj.status = OBJ_INIT;
+    this->obj.id = OBJ_ACTOR_ALLRANGE;
+    this->aiType = enemyIndex + AI360_10 + 3;
 
-    actor->obj.pos.x = gActors[10].obj.pos.x + D_i4_8019F4E4[arg1].x;
```

Comment:
For this level, defines for the missile actor slots seems like a good idea. we could then write AI360_10 + 3 in terms of those.